pkg/controller/importconfig/importconfig_controller.go

```diff
@@ -97,7 +96,17 @@ func (r *ReconcileImportConfig) Reconcile(ctx context.Context, request reconcile
 		return reconcile.Result{}, err
 	}
 
-	// make sure the managed cluster clusterrole, cluserrolebiding and bootstrap sa are updated
+	mode, err := helpers.DetermineKlusterletMode(managedCluster)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	worker, ok := r.workers[mode]
```

Comment:
how about use a factory instead of this? we will get the worker from the factory

may be
```
worker, err := getImportWorker(mode)
```